evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -369,3 +418,51 @@ def _transform_parameters(self, pipeline, proposed_parameters):
                 component_parameters["n_jobs"] = self.n_jobs
             parameters[name] = component_parameters
         return parameters
+
+    def _make_split_pipeline(self, estimator, pipeline_name=None):
+        basic_pipeline_parameters = {
+            "Select Columns Transformer": {"columns": self._selected_cols}
+        }
+        basic_pipeline = make_pipeline(
+            self.X,
+            self.y,
+            estimator,
+            self.problem_type,
+            sampler_name=self.sampler_name,
+            parameters=basic_pipeline_parameters,
+            extra_components=[SelectColumns],
+            extra_components_position="before_estimator",
+            use_estimator=False if self._selected_cat_cols else True,
+        )
+
+        if self._selected_cat_cols:
+            self._split = True
+            categorical_pipeline_parameters = {
+                "Select Columns Transformer": {"columns": self._selected_cat_cols}
+            }
+            categorical_pipeline = make_pipeline(
+                self.X,
+                self.y,
+                estimator,
+                self.problem_type,
+                sampler_name=self.sampler_name,
+                parameters=categorical_pipeline_parameters,
+                extra_components=[SelectColumns],
+                extra_components_position="before_preprocessing",
+                use_estimator=False,
+            )
+            input_pipelines = [basic_pipeline, categorical_pipeline]
```

Comment:
each sampler _should_ have the same parameters so theoretically it should be ok as it'll have the same outputs

Reply:
My question is whether sampling parts of the data independently is materially different than sampling the whole data once. For Undersampling, I think the answer is no but for oversampling I'm not sure. If the perf tests come back ok, then we can leave this question for a follow-up issue.

Reply:
Agreed here. Oversampling is handled differently for Categorical data versus Numeric data, so this could crunch out different results if done separately (SMOTE/SMOTEN vs SMOTENC) versus together. However, I do think the sampling strategy should be the same overall? In that, if the data was numeric + categorical, I think sampling with `SMOTE` and `SMOTEN` on the data separately uses the same techniques as sampling with `SMOTENC`, which should lead to very similar/same sampling results. 

Definitely something worth testing, however!